### PR TITLE
B0.20 b0.20.1

### PR DESCRIPTION
**Pull request by shadeofblue**

### CODE REVIEW

File: tests/golem/task/test_tasksession.py [hunk 1]
```diff
@@ -1217,8 +1217,11 @@ def test_ctd_is_none(self, mock_cat, *_):
     def test_multi_wtct(self, *_):
         # given
         self.msg = msg_factories.tasks.WantToComputeTaskFactory(num_subtasks=3)
-        ctd = msg_factories.tasks.ComputeTaskDefFactory(resources=None)
-        self.ts.task_manager.get_next_subtask.return_value = ctd
+
+        def ctd(*args, **kwargs):
+            return msg_factories.tasks.ComputeTaskDefFactory(resources=None)
+
+        self.ts.task_manager.get_next_subtask.side_effect = ctd
```
Comment (kubkon) [inline tests/golem/task/test_tasksession.py:1224]: Hmm, how come this was required here?

Comment (shadeofblue) [reply tests/golem/task/test_tasksession.py:1224]: @kubkon normally, CTD would be generated anew in each call to `get_next_subtask`... 

with the mock in place, the CTD is generated _once_ and then used as a return value for the mock and thus, iterations over subtasks operate on the same CTD and infuence each other... 

this change makes the CTD unique for each call again

Comment (kubkon) [reply tests/golem/task/test_tasksession.py:1224]: Hmm, I still can't see how that fixes it here. But that might be my lack of Python knowledge. So, wrapping

```python
msg_factories.tasks.ComputeTaskDefFactory(resources=None)
```
inside a separate scope (a function) causes a deep copy then?

**Review by jiivan — APPROVED** [review]
kawaczyherb

File: tests/golem/task/test_tasksession.py (hunk 1 above)
Comment (shadeofblue) [reply tests/golem/task/test_tasksession.py:1224]: @kubkon nah, it's a difference between passing a function and passing a value ;) ...

Comment (kubkon) [reply tests/golem/task/test_tasksession.py:1224]: OK, my misunderstanding, looks good!

**Pull request merged**